code/datums/components/regenerator.dm

```diff
@@ -0,0 +1,85 @@
+/**
+ * # Regenerator component
+ *
+ * A mob with this component will regenerate its health over time, as long as it has not received damage
+ * in the last X seconds. Taking any damage will reset this cooldown.
+ */
+/datum/component/regenerator
+	/// You will only regain health if you haven't been hurt for this many seconds
+	var/regeneration_delay
+	/// Health to regenerate per second
+	var/health_per_second
+	/// List of damage types we don't care about, in case you want to only remove this with fire damage or something
+	var/list/ignore_damage_types
+	/// Colour of regeneration animation, or none if you don't want one
+	var/outline_colour
+	/// If this is active we can't regenerate health
```

Comment:
What is "this"? This should be more obvious it's taking a timer id (though I don't know why you're doing that in the first place?)